components/experimental/src/units/ratio.rs

```diff
@@ -0,0 +1,228 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::{
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign},
+    str::FromStr,
+};
+
+use num_bigint::BigInt;
+use num_rational::Ratio;
+use num_traits::Signed;
+use num_traits::{One, Pow, Zero};
+use zerovec::ule::AsULE;
+
+use super::provider::{Base, ConversionInfoULE, SiPrefix, Sign};
+
+// TODO: implement AsULE for IcuRatio and use it in Data module.
+// TODO: add test cases for IcuRatio.
+/// A ratio type that uses `BigInt` as the underlying type.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct IcuRatio(Ratio<BigInt>);
+
+#[derive(Debug)]
+pub enum IcuRatioError {
+    /// Represents an error when parsing a `BigInt` from a string.
+    /// For example, the string "1/23" is invalid because it contains a '/' character.
+    ///              the string "3A" is invalid because it contains a non-numeric character.
+    BigIntParseError(num_bigint::ParseBigIntError),
+
+    /// Represents an error when parsing a ratio from a string.
+    /// For example, the string "1/0/2" is invalid because it contains more than one '/' character.
+    InvalidRatioString,
+}
+
+impl IcuRatio {
+    /// Creates a new `IcuRatio` from the given numerator and denominator.
+    pub fn new(numerator: BigInt, denominator: BigInt) -> Self {
+        Self(Ratio::new(numerator, denominator))
+    }
+
+    /// Extracts the factor ratio from `ConversionInfoULE`.
+    pub fn factor_from_conversion_info(conversion_info: &ConversionInfoULE) -> Self {
```

Comment:
Suggestion: these methods can _probably_ be `pub(crate)`; I don't think you need them in datagen

Reply:
sure, done